### ISSUE TITLE
Excessive dependencies

### ISSUE THREAD

**Issue by pgunn** [conda-forge/tensorflow-probability-feedstock#53]
The tensorflow-probability package doesn't actually need tensorflow, which pulls in a huge number of other things. Some packages that don't need tensorflow at all just need this; please consider following the pypi package's dependencies and not depending on tensorflow.

**Comment by jonas-eschle** [comment]
That's true, but the pip package allows for optional dependencies, which conda doesn't. The philosophy is therefore rather to include all dependencies, or rather more than less, see also this [discussion](https://conda.discourse.group/t/policy-recommendations-for-optional-dependencies/209) and this conda feature request: https://github.com/conda/conda/issues/7502

I am with you on this though, it might be worth to simply remove them now, as the user will anyway install TensorFlow or JAX to use TFP.

One remaining thing, the TFP team is a bit... stuck (weird?) with the recent releases, see 2.17, which is still not out (https://github.com/tensorflow/probability/issues/1832) and 2.18 also not coming (https://github.com/tensorflow/probability/issues/1851)
